ibm_cloud_sdk_core/utils.py

```diff
@@ -125,15 +134,17 @@ def read_from_credential_file(service_name, separator='='):
     return config
 
 def _parse_key_and_update_config(config, service_name, key, value):
-    if service_name in key:
-        index = key.find('_')
+    service_name = service_name.replace(' ', '_').replace('-', '_').lower()
+    key = key.replace(' ', '_').replace('-', '_').lower()
+    if key.startswith(service_name):
+        index = key.find(service_name)
```

Comment:
since we entered this block because `key` starts with `service_name`, you shouldn't need to do a find at all, right?
since the key starts with the service_name the index should always be 0 (or 1, whatever index the first character has).
Please see the logic in the CredentialUtils.parseCredentials() function:
https://github.com/IBM/java-sdk-core/blob/e3d5330b0d9e2ef9cf084cb7c6aa97d6ae4f1fb4/src/main/java/com/ibm/cloud/sdk/core/util/CredentialUtils.java#L352
Specifically this:
```
   serviceName = serviceName.toUpperCase().replaceAll("-", "_") + "_";
   ...
      if (key.startsWith(serviceName)) {
        String credentialName = key.substring(serviceName.length());
```
Note that this function changes the serviceName value to do the character replacement AND adds a `"_"` on the end so that the updated value of serviceName (e.g. `PERSONALITY_INSIGHTS_`) will serve as the actual "prefix" that we'd expect to see in property and env variable names.   Because we add the trailing `"_"` on the end, there's no `+1` needed when we compute the length of serviceName.

Reply:
Agree that I don't need to use find to get index, but how can we guarantee that the key would always be uppercase? I would prefer we transform both so that we perform the correct string matching